src/gretel_trainer/relational/core.py

```diff
@@ -806,13 +806,32 @@ def debug_summary(self) -> dict[str, Any]:
                         "parent_columns": key.parent_columns,
                     }
                 )
-            tables[table] = {
+            table_metadata = {
                 "column_count": len(self.get_table_columns(table)),
                 "primary_key": self.get_primary_key(table),
                 "foreign_key_count": this_table_foreign_key_count,
                 "foreign_keys": foreign_keys,
                 "is_invented_table": self._is_invented(table),
             }
+            if (
+                self.is_producer_of_invented_tables(table)
+                and (producer_metadata := self.get_producer_metadata(table)) is not None
+            ):
+                table_metadata["invented_table_details"] = {
+                    "table_type": "producer",
+                    "json_to_table_mappings": producer_metadata.table_name_mappings,
+                }
+            elif (
+                self._is_invented(table)
+                and (invented_table_metadata := self.get_invented_table_metadata(table))
+                is not None
+            ):
```

Comment:
You can remove the "first half" of both these conditionals, since they are essentially helper methods that under the hood perform the same check that the "second half" of each conditional check does (and maps it to a boolean).